android/app/src/main/kotlin/cc/retroshare/retroshare/MainActivity.kt

```diff
@@ -0,0 +1,140 @@
+
+
+package cc.retroshare.retroshare
+
+//import android.annotation.TargetApi
+//import android.os.Bundle
+//
+//import android.content.Intent
+//import android.os.Build
+//import io.flutter.plugin.common.MethodCall
+//
+//import androidx.annotation.NonNull
+import io.flutter.embedding.android.FlutterActivity
+import io.flutter.embedding.engine.FlutterEngine
+//import io.flutter.plugin.common.MethodChannel
+import io.flutter.plugins.GeneratedPluginRegistrant
+
+
+import android.annotation.TargetApi
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.os.Build
+import androidx.annotation.NonNull;
+import androidx.core.app.NotificationCompat
+
+//import io.flutter.embedding.engine.plugins.FlutterPlugin
+//import io.flutter.embedding.engine.plugins.activity.ActivityAware
+//import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding
+import io.flutter.plugin.common.BinaryMessenger
+import io.flutter.plugin.common.MethodCall
+import io.flutter.plugin.common.MethodChannel
+import io.flutter.plugin.common.MethodChannel.MethodCallHandler
+import io.flutter.plugin.common.MethodChannel.Result
+import io.flutter.plugin.common.PluginRegistry
+import io.flutter.plugin.common.PluginRegistry.Registrar
+
+
+//class MainActivity : FlutterActivity() {
+class MainActivity :  FlutterActivity() {
+    // Add for foreground project
+    private var methodChannel : MethodChannel? = null
+    private var permissionHandler: PermissionHandler? = null
+
+    companion object {
+        @JvmStatic
+        var notificationTitle: String? = "flutter_background foreground service"
+        @JvmStatic
+        var notificationText: String? = "Keeps the flutter app running in the background"
+        @JvmStatic
+        var notificationImportance: Int? = NotificationCompat.PRIORITY_DEFAULT
+
+        @JvmStatic
+        var notificationIconName: String? = "ic_notification_outerborder"
+        @JvmStatic
+        var notificationIconDefType: String? = "drawable"
+    }
+
+    override fun configureFlutterEngine(@NonNull flutterEngine: FlutterEngine) {
+        super.configureFlutterEngine(flutterEngine)
+        GeneratedPluginRegistrant.registerWith(flutterEngine)
```

Comment:
@selankon  I think you should also remove this line. Because `super.configureFlutterEngine(flutterEngine)` doing the same. work as  `GeneratedPluginRegistrant.registerWith(flutterEngine)` is doing.

Reply:
After removing this line everything started working.

Reply:
Awesome. So now is working?

Reply:
yes.